### PR TITLE
docs: rewrite with getting started

### PR DESCRIPTION
**Pull request by philsturgeon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,14 +37,6 @@ docker run -P stoplight/prism:3 mock -h 0.0.0.0 api.oas2.yml
 - [ ] Recording / "Learning" mode to create spec files
 - [ ] Data Persistence (allow Prism act like a sandbox)
 
-## Experimental programmatic APIs (advanced topic)
```
Comment (mallachari) [inline README.md]: We won't expose this API any more?

File: docs/guides/cli.md [hunk 2]
```diff
@@ -51,5 +52,8 @@ Connection: keep-alive
 
 The body, headers, etc. for this response will be taken from the API description document.
 
+## Running in Production
```
Comment (pytlesk4) [inline docs/guides/cli.md]: Why would I run prism in production?

File: docs/getting-started/installation.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+
+# Installation
```
Comment (pytlesk4) [inline docs/getting-started/installation.md]: Can you add an update prism section?

File: docs/guides/mock-responses.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    "name": "doggie",
+    "photoUrls": [
+        "string"
+    ]
+}
+```
+
+Notice that `name` had an `example` with a value so Prism used it, but `photoUrls` did not, so it just returned `"string"`. 🤷‍♂️
+
+### Dynamic Response Generation
```
Comment (pytlesk4) [inline docs/guides/mock-responses.md]: Not rendering correctly

image

File: docs/guides/workflow.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Workflow
```
Comment (pytlesk4) [inline docs/guides/workflow.md]: Feels like an introduction?

File: docs/guides/cli.md (hunk 2 above)
Comment (philsturgeon) [reply docs/guides/cli.md]: People do all sorts of things and I'd rather have an example for them than not. I'm moving this from a nested README to a smarter place, and we can improve on its explanation another time.

File: docs/getting-started/installation.md (hunk 3 above)
Comment (philsturgeon) [reply docs/getting-started/installation.md]: Update is just installing again right?

File: docs/guides/workflow.md (hunk 5 above)
Comment (philsturgeon) [reply docs/guides/workflow.md]: It's introductory in the sense that it teaches people specifically about where it fits into the grand scheme of things, which you might not need to be told if you are already very aware of the API design-first workflow. I put it in as a guide for people who are not.

File: docs/guides/mock-responses.md (hunk 4 above)
Comment (philsturgeon) [reply docs/guides/mock-responses.md]: Fixed! Great spot, this is a really weird whitespace bug which I'll make an issue for.

File: README.md (hunk 1 above)
Comment (philsturgeon) [reply README.md]: It was a huge oversight on my part that this API ever made it in. We were not meant to be advertising the JS API until it was stable, and we were meant to be explaining it was experimental in the `packages/http/README.md` not the _main README_. Putting this here before it is ready (#511) is an advertisement we don't want.

Hopefully not too many people have started building anything in production around it.

File: docs/getting-started/installation.md (hunk 3 above)
Comment (philsturgeon) [reply docs/getting-started/installation.md]: > Note, the binaries do _not_ auto-update, so you will need to run it again to install new versions.

We got it covered :)

File: docs/guides/cli.md (hunk 2 above)
Comment (XVincentX) [reply docs/guides/cli.md]: Prims will run in production on the hosted platform (Stoplight), for example. I want *you* to know how to use it properly :trollface:

**Pull request merged**